src/pages/SubscriptionsPage.tsx

```diff
@@ -0,0 +1,292 @@
+import { useMemo } from 'react'
+import { generatePath, useSearchParams } from 'react-router-dom'
+
+import { InfiniteScroll, Status, StatusType } from '~/components/designSystem'
+import {
+  Filters,
+  formatFiltersForSubscriptionQuery,
+  SubscriptionAvailableFilters,
+} from '~/components/designSystem/Filters'
+import { SearchInput } from '~/components/SearchInput'
+import { SubscriptionsList } from '~/components/subscriptions/SubscriptionsList'
+import { TimezoneDate } from '~/components/TimezoneDate'
+import { SUBSCRIPTION_LIST_FILTER_PREFIX } from '~/core/constants/filters'
+import { getIntervalTranslationKey } from '~/core/constants/form'
+import { CustomerSubscriptionDetailsTabsOptionsEnum } from '~/core/constants/tabsOptions'
+import { CUSTOMER_SUBSCRIPTION_DETAILS_ROUTE } from '~/core/router'
+import { StatusTypeEnum, Subscription, useGetSubscriptionsListLazyQuery } from '~/generated/graphql'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+import { useDebouncedSearch } from '~/hooks/useDebouncedSearch'
+import { PageHeader } from '~/styles'
+
+gql`
+  fragment SubscriptionForSubscriptionsList on Subscription {
+    id
+    status
+    startedAt
+    nextSubscriptionAt
+    nextSubscriptionType
+    name
+    nextName
+    externalId
+    subscriptionAt
+    endingAt
+    terminatedAt
+    customer {
+      id
+      name
+      displayName
+      applicableTimezone
+    }
+    plan {
+      id
+      isOverridden
+      amountCurrency
+      name
+      interval
+    }
+    nextPlan {
+      id
+      name
+      code
+      interval
+    }
+    nextSubscription {
+      id
+      name
+      externalId
+      status
+    }
+  }
+
+  query getSubscriptionsList(
+    $limit: Int
+    $page: Int
+    $searchTerm: String
+    $status: [StatusTypeEnum!]
+    $externalCustomerId: String
+    $overriden: Boolean
+    $planCode: String
+  ) {
+    subscriptions(
+      limit: $limit
+      page: $page
+      status: $status
+      searchTerm: $searchTerm
+      externalCustomerId: $externalCustomerId
+      overriden: $overriden
+      planCode: $planCode
+    ) {
+      collection {
+        ...SubscriptionForSubscriptionsList
+      }
+      metadata {
+        currentPage
+        totalPages
+        totalCount
+      }
+    }
+  }
+`
+
+const SubscriptionsPage = () => {
+  const { translate } = useInternationalization()
+  const [searchParams] = useSearchParams()
+
+  const filtersForSubscriptionQuery = useMemo(() => {
+    return formatFiltersForSubscriptionQuery(searchParams)
+  }, [searchParams])
+
+  const [getSubscriptions, { data, error, loading, variables, fetchMore }] =
+    useGetSubscriptionsListLazyQuery({
+      variables: {
+        limit: 20,
+        ...filtersForSubscriptionQuery,
+      },
+    })
+
+  const { debouncedSearch, isLoading } = useDebouncedSearch(getSubscriptions, loading)
+
+  const subscriptions = data?.subscriptions.collection as Subscription[]
+  const hasSearchParams =
```

Comment:
I wonder how not check individually each params. If we add one search param later we may miss updating this condition maybe?
Could that be done differently?